proggy.cpp

```diff
@@ -120,11 +169,24 @@ void customerPrinter(std::vector<Customer> const& vec) {
 
 }
 
-std::vector<Customer> const applyDiscount(std::vector<Customer> vec, std::string comp){
+
+// std::vector<Customer> const applyDiscount(std::vector<Customer> vec, std::string comp){
+void applyDiscount(std::vector<Customer>& vec, const std::string comp){
+//** KOOS: 
+//** vector should be passed by ref and modified directly else a copy will be created
+//** as input and another copy as output..
+
+
     auto ret = vec;
-    for (auto &it : vec ) {
-        ret.push_back(applyDiscount(it, comp));
+    
+    for (Customer& it : vec ) {
+        //ret.push_back(applyDiscount(it, comp));
+//**KOOS:
+//** This re-inserts a duplicate record into the vector
```

Comment:
yep! Nice catch! This was the last change I made in the program to "simplify" it a bit. Better testing would have caught this.